packages/core-utils/src/query.js

```diff
@@ -466,8 +466,19 @@ export function getRoutingParams(config, currentQuery, ignoreRealtimeUpdates) {
   // Replace FLEX placeholder with OTP flex modes
   if (params.mode) {
     // Ensure query is in reduced format to avoid replacing twice
-    const mode = reduceOtpFlexModes(params.mode.split(",")).join(",");
-    params.mode = mode.replace("FLEX", "FLEX_EGRESS,FLEX_ACCESS,FLEX_DIRECT");
+    const mode = reduceOtpFlexModes(params.mode.split(","));
+    params.mode = mode
+      .map(m => {
+        // If both the expanded and shrunk modes are included, remove the exapnded one
+        if (m === "FLEX_EGRESS" || m === "FLEX_ACCESS" || m === "FLEX_DIRECT") {
+          if (mode.includes("FLEX")) return "";
+        }
+        if (m === "FLEX") {
+          return "FLEX_EGRESS,FLEX_ACCESS,FLEX_DIRECT";
+        }
+        return m;
+      })
+      .join(",");
```

Comment:
You should extract this changed block into its own method and add a test so we can see the expected inputs/outputs.

Reply:
Good call! Lots of unit tests added in 70d83757ca12f50aea4417d847732854c6c660b8